### PR TITLE
Fix accessors to right hand side of a constraint.

### PR DESCRIPTION
**Pull request by florent-lamiraux**

### CODE REVIEW

**Comment by jmirabel** [comment]
Thanks for the unit-test. If you are done with testing, you can merge.

File: tests/solver-by-substitution.cc [hunk 1]
```diff
@@ -846,3 +850,82 @@ BOOST_AUTO_TEST_CASE (rightHandSide)
...
+  Transform3f tf1; // Identity
+  vector3_t u; u << 0, -.2, 0;
+  Transform3f tf2; tf2.translation (u);
+  ImplicitPtr_t c1 (hpp::constraints::implicit::RelativePose::create
+                    ("RelativeTransformation", device, ee1, ee2, tf1, tf2,
+                     std::vector <bool> (6, true), comp1));
+  u << 1.2, 0, -1;
+  tf2.translation (u);
+  ImplicitPtr_t c2 (hpp::constraints::explicit_::RelativePose::create
+                    ("Transformation", device, JointPtr_t (), ee1, tf2, tf1,
```
Comment (jmirabel) [inline tests/solver-by-substitution.cc]: I have a SEGV in this test. I think it comes from this line as the constructor arguments do not make sense to me. @florent-lamiraux Can you confirm this ?

Comment (jmirabel) [reply tests/solver-by-substitution.cc]: I precise that I am trying this with Ubuntu 18.04.

Comment (florent-lamiraux) [reply tests/solver-by-substitution.cc]: The arguments seem correct to me. I will test in ubuntu-16.04.

Comment (jmirabel) [reply tests/solver-by-substitution.cc]: I don't understand how this function can be explicit. How does the function knows what SE3 joint values are the output ?

Comment (florent-lamiraux) [reply tests/solver-by-substitution.cc]: You are right. It is not correct. It seems that we have never tested this in debug mode. I will fix the issue in a future pull request.

**Pull request merged**